packages/venia-concept/src/components/Checkout/__tests__/form.spec.js

```diff
@@ -37,114 +43,171 @@ const defaultProps = {
 
 beforeEach(() => {
     mockCancelCheckout.mockReset();
-    mockEditOrder.mockReset();
     mockSubmitShippingAddress.mockReset();
     mockSubmitOrder.mockReset();
     mockSubmitPaymentMethodAndBillingAddress.mockReset();
     mockSubmitShippingMethod.mockReset();
 });
 
 test('renders an overview Form component if not editing', () => {
-    const component = testRenderer.create(<Form {...defaultProps} />);
+    const component = createTestInstance(<Form {...defaultProps} />);
 
     expect(component.toJSON()).toMatchSnapshot();
 });
 
-test('renders an editable Form component if editing', () => {
-    const props = {
-        ...defaultProps,
-        editing: 'address'
-    };
-    const component = testRenderer.create(<Form {...props} />);
+test('dismissing the form cancels checkout', () => {
+    const component = createTestInstance(<Form {...defaultProps} />);
 
-    expect(component.toJSON()).toMatchSnapshot();
+    act(() => {
+        component.root.findAllByType(Button)[0].props.onClick();
+    });
+
+    expect(mockCancelCheckout).toHaveBeenCalled();
 });
 
-test('dismissCheckout instance function calls props.cancelCheckout', () => {
-    const component = testRenderer.create(<Form {...defaultProps} />);
+test('clicking address form edit sets `editing` state value to "address"', () => {
+    const mockSetEditing = jest.fn();
+    useState.mockReturnValueOnce([null, mockSetEditing]);
 
-    component.root.findAllByType(Button)[0].props.onClick();
+    const component = createTestInstance(<Form {...defaultProps} />);
 
-    expect(mockCancelCheckout).toHaveBeenCalled();
+    act(() => {
+        component.root.findAllByType(Section)[0].props.onClick();
```

Comment:
Here, an implementation detail of a different component is being tested. `Form` passes `setEditing` to `EditableForm` without any knowledge that it'll end up being a click handler, yet this test drills down even further into `EditableForm` to find a particular child (again, unknown) and calls its click handler.

All this test should do is verify that `setEditing` is passed to its children. We don't need to mock `useState` or verify that `setEditing` updates state.

Reply:
Yea so what I ended up doing was not writing tests for each of the sub-components that were created in this PR. Instead I just mocked React and wrote tests that could apply to `EditableForm` and `Overview`.

I have a few options:
1) Split all these tests into individual files
2) Remove them from this test entirely.
3) Leave it as is.

What would you like me to do?

Reply:
Definitely split the tests into files that correspond to one component each. We're only writing unit tests.

Most of these are simple enough that a snapshot test should get you 100% coverage.